pkg/validation/validation.go

```diff
@@ -25,56 +25,94 @@ func NewBuilder() ValidationBuilder {
 // Validator knows how to validate UpgradeConfig CRs.
 //go:generate mockgen -destination=mocks/mockValidation.go -package=mocks github.com/openshift/managed-upgrade-operator/pkg/validation Validator
 type Validator interface {
-	IsValidUpgradeConfig(uC *upgradev1alpha1.UpgradeConfig, cV *configv1.ClusterVersion, logger logr.Logger) (bool, error)
+	IsValidUpgradeConfig(uC *upgradev1alpha1.UpgradeConfig, cV *configv1.ClusterVersion, logger logr.Logger) (ValidatorResult, error)
 }
 
 type validator struct{}
 
-func (v *validator) IsValidUpgradeConfig(uC *upgradev1alpha1.UpgradeConfig, cV *configv1.ClusterVersion, logger logr.Logger) (bool, error) {
+type ValidatorResult struct {
+	IsRollBack bool
```

Comment:
Where is this used? What is its purpose?

Reply:
Good spot.
There was a scenario of `false, nil` for roll back which I tried to capture.
Ok to remove iyo?

Reply:
Will we support rollback? Do we need to separate this condition out?

Reply:
@bmeng @dofinn I'm going to remove this